virttest/utils_stress.py

```diff
@@ -86,8 +111,49 @@ def vm_stress_events(self, event, vm):
                                 'cur_config': self.current_vcpu,
                                 'cur_live': self.current_vcpu,
                                 'guest_live': self.current_vcpu}
-                    utils_hotplug.check_vcpu_value(vm, exp_vcpu, option="--live")
+                    utils_hotplug.check_vcpu_value(
+                        vm, exp_vcpu, option="--live")
             elif "reboot" in event:
                 vm.reboot()
+            elif "nethotplug" in event:
+                for iface_num in range(int(self.iface_num)):
+                    logging.debug("Try to attach interface %d" % iface_num)
+                    mac = utils_net.generate_mac_address_simple()
+                    options = ("%s %s --model %s --mac %s %s" %
+                               (self.iface_type, self.iface_source['network'],
+                                self.iface_model, mac, self.attach_option))
+                    logging.debug("VM name: %s , Options for Network attach: %s", vm.name, options)
+                    ret = virsh.attach_interface(vm.name, options,
+                                                 ignore_status=True)
+                    time.sleep(self.event_sleep_time)
+                    libvirt.check_exit_status(ret)
+                    if self.detach_option:
+                        options = ("--type %s --mac %s %s" %
+                                   (self.iface_type, mac, self.detach_option))
+                        logging.debug("VM name: %s , Options for Network detach: %s", vm.name, options)
+                        ret = virsh.detach_interface(vm.name, options,
+                                                     ignore_status=True)
+                        libvirt.check_exit_status(ret)
+            elif "diskhotplug" in event:
+                for disk_num in range(len(self.device_source_names)):
+                    disk = {}
+                    disk_attach_error = False
+                    disk_name = os.path.join(self.path, vm.name, self.device_source_names[disk_num])
+                    device_source = libvirt.create_local_disk(
+                        self.disk_type, disk_name, self.disk_size, disk_format=self.disk_format)
+                    disk.update({"format": self.disk_format,
+                                 "source": device_source})
+                    disk_xml = Disk(self.disk_type)
+                    disk_xml.device = self.disk_device
+                    disk_xml.driver = {"name": self.disk_driver, "type": self.disk_format}
+                    attach_option = ""
+                    if self.attach_option:
+                        attach_option = self.attach_option
+                    ret = virsh.attach_disk(vm.name, disk["source"], self.device_target[disk_num], attach_option, debug=True)
```

Comment:
attach_option is not necessary here, as we initialize `self.attach_option` with `""` as default in L36
directly using it will be the same right ?

Reply:
attach_option is required to provide various disk attach options [like attach type, targetbus, wwn explicit driver overriding, disk type etc.]  It is not mandatory, so the default value none. But user can provide required options via configuration file.